ls: Migrate away from DeprecatedFile

BenWiederhake opened the pull request:
Note that since many low-level bare C APIs are used, null-terminated strings are still necessary in many places, which sadly required the addition of many DeprecatedStrings.

Migrating all of these to `Core::Account` etc. seems a bit too much, so here I focus only on `DeprecatedFile`.

Advances #17129, unblocks #18812.

ADKaster commented on Userland/Utilities/ls.cpp:
```diff
@@ -274,8 +274,8 @@ static size_t print_name(const struct stat& st, DeprecatedString const& name, ch
+        if (path_for_link_resolution.has_value()) {
+            auto link_destination_or_error = FileSystem::read_link(path_for_link_resolution.value());
             if (link_destination_or_error.is_error()) {
                 perror("readlink");
```
This is subtly broken, instead of calling `perror("readlink")`, now we must `warnln("{}", link_destination_or_error.error())` as the FileSystem API will not set errno but return the Error directly.

BenWiederhake replied:
This makes me want to write a fault-injector, perhaps using ptrace or something, just to see where else things go wrong.

ADKaster reviewed (changes requested):
This generally looks like an improvement. The conversion of the rest of the file can definitely wait for later. One bug though:

BenWiederhake commented:
Changes since last push:
- Rebased for convenience
- No longer read from (possibly-overwritten-or-never-set) `errno`, instead use the `AK::Error` for error printing.
